src/main/java/com/google/sps/app/servlets/SignUp.java

```diff
@@ -35,17 +39,25 @@ public class SignUp extends HttpServlet{
     public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
         String username = request.getParameter(USER_NAME_PARAM).trim();
         
-        // if(!Validator.validUserName(username)) {
-        //     response.getWriter().println("Invalid Username: Please only use letters and numbers.");
-        //     return;
-        // } else if (!Validator.userNameAvalible(username)) {
-        //     response.getWriter().println("Username Taken: <a href=\"/LogIn\">Login</a>");
-        //     return;
-        // }
+        // gaurd clause for invalid usernames
+        String error = Validator.validateUserName(username);
+        if(error != "") {
+            error = error == "length" ? USERNAME_LENGTH : IMPROPER_CHARACTERS;
```

Comment:
Same here with ==

Reply:
Sorry I must have changed it in login. Now it should be done everywhere